programs/account-compression/src/state/batched_queue.rs

```diff
@@ -0,0 +1,827 @@
+                    num_stores,
+                    num_hashchain_stores,
+                    account_data,
+                )?;
+            Ok(ZeroCopyBatchedQueueAccount {
+                account,
+                batches,
+                value_vecs,
+                bloom_filter_stores,
+                hashchain_store,
+            })
+        }
+    }
+
+    pub fn init(
+        metadata: QueueMetadata,
+        num_batches_output_queue: u64,
+        output_queue_batch_size: u64,
+        output_queue_zkp_batch_size: u64,
+        account_data: &mut [u8],
+        num_iters: u64,
+        bloom_filter_capacity: u64,
+    ) -> Result<ZeroCopyBatchedQueueAccount> {
+        let account = bytes_to_struct_checked::<BatchedQueueAccount, true>(account_data)?;
+        unsafe {
+            (*account).init(
+                metadata,
+                num_batches_output_queue,
+                output_queue_batch_size,
+                output_queue_zkp_batch_size,
+                bloom_filter_capacity,
+            )?;
+
+            let (batches, value_vecs, bloom_filter_stores, hashchain_store) = init_queue(
+                &(*account).queue,
+                (*account).metadata.queue_type,
+                account_data,
+                num_iters,
+                bloom_filter_capacity,
+                &mut 0,
+            )?;
+            Ok(ZeroCopyBatchedQueueAccount {
+                account,
+                batches,
+                value_vecs,
+                bloom_filter_stores,
+                hashchain_store,
+            })
+        }
+    }
+
+    pub fn insert_into_current_batch(&mut self, value: &[u8; 32]) -> Result<()> {
+        let current_index = self.get_account().next_index;
+        unsafe {
+            insert_into_current_batch(
+                (*self.account).metadata.queue_type,
+                &mut (*self.account).queue,
+                &mut self.batches,
+                &mut self.value_vecs,
+                &mut self.bloom_filter_stores,
+                &mut self.hashchain_store,
+                value,
+                None,
+                Some(current_index),
+            )?;
+            (*self.account).next_index += 1;
+        }
+        Ok(())
+    }
+
+    /// Zero out a leaf by index if it exists in the queues value vec. If
+    /// checked fail if leaf is not found.
+    pub fn prove_inclusion_by_index_and_zero_out_leaf(
+        &mut self,
+        leaf_index: u64,
+        value: &[u8; 32],
+    ) -> Result<()> {
+        for (batch_index, batch) in self.batches.iter().enumerate() {
+            if batch.value_is_inserted_in_batch(leaf_index)? {
+                let index = batch.get_value_index_in_batch(leaf_index)?;
+                let element = self.value_vecs[batch_index]
+                    .get_mut(index as usize)
+                    .ok_or(AccountCompressionErrorCode::InclusionProofByIndexFailed)?;
+
+                if element == value {
+                    *element = [0; 32];
```

Comment:
```suggestion
```
We need to remove this so that we can use the value from `output_queue.value_vec[full_batch_index]` directly instead from the indexer.